test-engineering/contract-tests/client/tests/test_merino.py

```diff
@@ -74,48 +153,12 @@ def assert_200_response(
             continue
 
 
-def test_merino(merino_url: str, steps: List[Step], kinto_icon_urls: Dict[str, str]):
+def test_merino(steps: List[Step], step_functions: Dict[Service, Callable]):
     """Test for requesting suggestions from Merino."""
 
     for step in steps:
         # Each step in a test scenario consists of a request and a response.
         # Use the parameters to perform the request and verify the response.
 
-        method = step.request.method
-        url = f"{merino_url}{step.request.path}"
-        headers = {header.name: header.value for header in step.request.headers}
-
-        r = requests.request(method, url, headers=headers)
-
-        error_message = (
-            f"Expected status code {step.response.status_code},\n"
-            f"but the status code in the response from Merino is {r.status_code}.\n"
-            f"The response content is '{r.text}'."
-        )
-
-        assert r.status_code == step.response.status_code, error_message
-
-        if r.status_code == 200:
-            # If the response status code is 200 OK, use the
-            # assert_200_response() helper function to validate the content of
-            # the response from Merino. This includes creating a pydantic model
-            # instance for checking the field types and comparing a dict
-            # representation of the model instance with the expected response
-            # content for this step in the test scenario.
-            assert_200_response(
-                step_content=step.response.content,
-                merino_content=ResponseContent(**r.json()),
-                kinto_icon_urls=kinto_icon_urls,
-            )
-            continue
-
-        if r.status_code == 204:
-            # If the response status code is 204 No Content, load the response content
-            # as text and compare against the value in the response model.
-            assert r.text == step.response.content
-            continue
-
-        # If the request to Merino was not successful, load the response
-        # content into a Python dict and compare against the value in the
-        # response model
-        assert r.json() == step.response.content
+        step_function: Callable = step_functions[step.request.service]
```

Comment:
I'm wondering if it's helpful to have a type annotation on this given that `step_functions` is annotated values can only be objects of type `Callable`. So it's redundant. 🤔 

Happy to leave this in if you think it's helps comprehensibility of the code.

Reply:
change applied: https://github.com/mozilla-services/merino/pull/409/commits/670814d39d81946a96d7d0e48d99c0e89adce4fd